pkg/handler3/handler.go

```diff
@@ -207,16 +266,26 @@ func (o *OpenAPIService) RegisterOpenAPIV3VersionedService(servePath string, han
 	return nil
 }
 
-func (o *OpenAPIV3Group) UpdateSpec(specBytes []byte) (err error) {
+func (o *OpenAPIV3Group) UpdateSpec(openapi *spec3.OpenAPI) (err error) {
 	o.rwMutex.Lock()
 	defer o.rwMutex.Unlock()
 
 	o.pbCache = o.pbCache.New(func() ([]byte, error) {
-		return ToV3ProtoBinary(specBytes)
+		json, err := o.jsonCache.Get()
```

Comment:
Do you know for sure what's going to be captured in that closure? `o` or `o.jsonCache`? Because if it's the latter, `o.jsonCache` is made obsolete line 280 below, so that's fishy. Can we topologically sort these instantiations just to be safe (and the intent actually looks a little better).